sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1309,9 +1309,9 @@ object CodeGenerator extends Logging {
   // bytecode instruction
   final val MUTABLESTATEARRAY_SIZE_LIMIT = 32768
 
-  // The java source code generated by whole codegen is compiled on the Driver side and sent to
-  // each Executor as the execution code to process data. This is very effective in processing
-  // large amounts of data in a distributed environment. However, in the test environment,
+  // The Java source code generated by whole-stage codegen on the Driver side is sent to each
```

Comment:
So it's generated on the driver and compiled on the executor, you're sure? seems OK

Reply:
Yes. Although compilation occurs on the driver too, it's just used to check.

Reply:
LGTM, thank you very much!

Reply:
Thanks for you remind too.